src/Http/Routing/src/EndpointRoutingMiddleware.cs

```diff
@@ -102,6 +108,71 @@ private Task SetRoutingAndContinue(HttpContext httpContext)
             }
 
             Log.MatchSuccess(_logger, endpoint);
+
+            var shortCircuitMetadata = endpoint.Metadata.GetMetadata<ShortCircuitMetadata>();
+            if (shortCircuitMetadata is not null)
+            {
+                // This check should be kept in sync with the one in EndpointMiddleware
+                if (!_routeOptions.SuppressCheckForUnhandledSecurityMetadata)
+                {
+                    if (endpoint.Metadata.GetMetadata<IAuthorizeData>() is not null)
+                    {
+                        ThrowCannotShortCircuitAnAuthRouteException(endpoint);
+                    }
+
+                    if (endpoint.Metadata.GetMetadata<ICorsMetadata>() is not null)
+                    {
+                        ThrowCannotShortCircuitACorsRouteException(endpoint);
+                    }
+                }
+
+                if (shortCircuitMetadata.StatusCode.HasValue)
+                {
+                    httpContext.Response.StatusCode = shortCircuitMetadata.StatusCode.Value;
+                }
+
+                if (endpoint.RequestDelegate is not null)
```

Comment:
Else log: The request is being short circuited without running the remainder of the pipeline.